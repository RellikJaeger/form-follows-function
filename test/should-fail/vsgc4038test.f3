/*
* Regression test:  An exception has occurred in the F3 compiler
*
* @compilearg -sourcepath
* @compilearg test/should-fail
* @test/compile-error
*/
public class vsgc4038test {
var test:vsgc4038sub;
public function func() {
test.f;
1
}
}
