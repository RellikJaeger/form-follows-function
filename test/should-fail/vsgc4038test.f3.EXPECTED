test/should-fail/vsgc4038test.f3:10: Note: Need explicit type for func because of cycle in definition.
public function func() {
^
test/should-fail/vsgc4038sub.f3:6: This is the cyclic reference to func that prevents type inference.
public var f=function(){vsgc4038test{}.func()}
                                      ^
1 error
